substrate/frame/support/procedural/src/pallet/expand/pallet_struct.rs

```diff
@@ -246,6 +246,14 @@ pub fn expand_pallet_struct(def: &mut Def) -> proc_macro2::TokenStream {
 						implemented by the runtime")
 			}
 
+			fn name_hash() -> [u8; 16] {
+				<
+					<T as #frame_system::Config>::PalletInfo as #frame_support::traits::PalletInfo
+				>::name_hash::<Self>()
+					.expect("Pallet is part of the runtime because pallet `Config` trait is \
+						implemented by the runtime")
```

Comment:
So, this might not be true under a misconfiguration, i.e. someone implements the pallet `Config` trait for a runtime, but forgets to include the actual pallet in `construct_runtime`.

Hopefully we do now have lints or warnings to catch this, but on the off-chance that we don't, I'd want to make sure that this `expect` doesn't trigger and cause a panic in runtime.

Reply:
This assumption is already used there multiple times.